### DISCUSSION TITLE
GUI Support?

### DISCUSSION THREAD

**Discussion by sshsphere** [Ircama/epson_print_conf#5] [Ideas]
Hey!
This is a really nice tool :) ! But it would be even nicer if it had an easy to use UI so the average person could use it. That would be a much better option for everyone than the paid and/or virus ridden "Epson Adjustment" software currently available online.

No guarantees but I might make one and send a PR if I have the time. Is that something you would be interested in?

**Comment by dgerber** [comment]
Agreed, that's why I started adding a GUI (to reinkpy).

But I think it would have to include autodiscovery — if a user can find
and enter their device IP, probably they can use the CLI too.

**Reply by sshsphere** [reply]
I actually already made a script for finding printer names and IPs: https://gist.github.com/sshsphere/fa42b8d1d8a177709ed4d65df580c127

Although I currently don't know of a reliable way to automatically select the correct printer model in code. I imagine it shouldn't be much trouble for the user to select the printer model they have themselves.

**Comment by dgerber** [comment]
> I actually already made a script for finding printer names and IPs: https://gist.github.com/sshsphere/fa42b8d1d8a177709ed4d65df580c127

Good, it looks a bit simpler than "SNMP-discovery"!

> Although I currently don't know of a reliable way to automatically select the correct printer model in code. I imagine it shouldn't be much trouble for the user to select the printer model they have themselves.

I've just pushed some notes for SNMP here: <https://codeberg.org/atufi/reinkpy/src/branch/feature-snmp>

The GUI does automatic model selection.

**Comment by Ircama** [comment]
While I have some experience with [wxPython](https://wxpython.org/index.html#) and [tkinter](https://docs.python.org/3/library/tkinter.html), I think a GUI is overkilling here. You can get the status of a printer via `python3 epson_print_conf.py -m XP-205 -a 192.168.1.87` (printer model and IP address) and reset the waste ink levels via `python3 epson_print_conf.py -m XP-205 -a 192.168.1.87 --reset_waste_ink` (again printer model and IP address; also, this command is very rarely used). Is it worth designing a UI for such simple use cases? Besides, *epson_print_conf.py* allows a number of additional options and filters for specific usage and an API interface. Really a GUI is needed?

**Reply by sshsphere** [reply]
I think so. The average person doesn't know what "python" and "pip" is and probably never opened a command terminal in their life. I imagine making an exe, even if its just two buttons for the most common use case (reset waste ink) would help a lot of people

**Reply by Ircama** [reply]
OK, I see now, depending on free time (my initial target was to produce a technical component anyway). A good tool to make an exe is https://pyinstaller.org.

I of course accept contributions.

**Reply by sshsphere** [reply]
Created a (draft) PR [here](https://github.com/Ircama/epson_print_conf/pull/6), lemme know what you think so far

**Reply by Ircama** [reply]
Thanks! I'll surely check it as soon as I have free time
